packages/trader/src/Stores/Modules/CFD/cfd-store.js

```diff
@@ -298,16 +345,55 @@ export default class MT5Store extends BaseStore {
     }
 
     @action.bound
-    async topUpVirtual() {
+    async topUpVirtual(platform) {
         this.root_store.ui.setTopUpInProgress(true);
-        const response = await WS.authorized.mt5Deposit({
-            to_mt5: this.current_account.login,
-        });
+        let response;
+
+        switch (platform) {
+            case 'dxtrade': {
+                response = await WS.authorized.send({
+                    trading_platform_deposit: 1,
+                    platform: 'dxtrade',
+                    to_account: this.current_account.account_id,
+                });
+                break;
+            }
+            case 'mt5': {
+                response = await WS.authorized.mt5Deposit({
+                    to_mt5: this.current_account.login,
+                });
+                break;
+            }
+            default: {
+                response.error = 'Invalid platform';
+                break;
+            }
+        }
+
         if (!response.error) {
-            await WS.authorized.mt5LoginList().then(this.root_store.client.responseMt5LoginList);
-            const new_balance = this.root_store.client.mt5_login_list.find(
-                item => item.login === this.current_account.login
-            ).balance;
+            let new_balance;
+            switch (platform) {
+                case 'dxtrade': {
+                    await WS.authorized
+                        .tradingPlatformAccountsList('dxtrade')
+                        .then(this.root_store.client.responseTradingPlatformAccountsList);
+                    new_balance = this.root_store.client.dxtrade_accounts_list.find(
+                        item => item.account_id === this.current_account.account_id
+                    )?.balance;
+                    break;
+                }
+                case 'mt5': {
+                    await WS.authorized.mt5LoginList().then(this.root_store.client.responseMt5LoginList);
+
+                    new_balance = this.root_store.client.mt5_login_list.find(
+                        item => item.login === this.current_account.login
+                    )?.balance;
+                    break;
+                }
+                default: {
+                    break;
+                }
```

Comment:
the empty `default` section is not necessary in switch/case when there is no default behavior

Reply:
@mahdiaryayi-fs `eslint` throws warning if we don't have default block in switch case 😅